### DISCUSSION TITLE
Usage with RxFire and Typescript?

### DISCUSSION THREAD

**Discussion by JanErikFoss** [invertase/react-native-firebase#5502] [Q&A]
How am I supposed to use this with RxFire and Typescript?
The types used by RNFirebase are not the same as used by the firebase js sdk, so typescript is yelling at me when I'm trying to pass them in to RxFire.

As a workaround I can delete _index.d.ts_ from the firebase js sdk package in node_modules, which turns all types from that package into _any_ types.
However this is obviously not an ideal solution.

**Comment by JanErikFoss** [comment]


**Comment by mikehardy** [comment]
I've never used RxFire, but the general idea is that our types should conform to the firebase-js-sdk *where possible* (there are some small differences we can't bridge, like we have to return Promise for some methods that are sync in firebase-js-sdk because of the nature of react-native, the JS/Native bridge is asynchronous and simply won't allow sync calls in some areas.

Other than that, if there are some types that are not conformant and you want to propose a PR we can collaborate and get it merged
